sycl/test/host-interop-task/host-task.cpp

```diff
@@ -0,0 +1,125 @@
+  S::buffer<int, 1> Buf2;
+  std::mutex Mutex;
+  std::condition_variable CV;
+};
+
+void Thread1Fn(Context &Ctx) {
+  // 0. initialize resulting buffer with apriori wrong result
+  {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::host_buffer>
+        Acc(Ctx.Buf2);
+
+    for (size_t Idx = 0; Idx < Acc.get_count(); ++Idx)
+      Acc[Idx] = -1;
+  }
+
+  // 1. submit task writing to buffer 1
+  Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::global_buffer>
+        GeneratorAcc(Ctx.Buf1, CGH);
+
+    auto GeneratorKernel = [GeneratorAcc]() {
+      for (size_t Idx = 0; Idx < GeneratorAcc.get_count(); ++Idx)
+        GeneratorAcc[Idx] = Idx;
+    };
+
+    CGH.single_task<class GeneratorTask>(GeneratorKernel);
+  });
+
+  // 2. submit host task writing from buf 1 to buf 2
+  Ctx.Queue.submit([&](S::handler &CGH) {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::host_buffer>
+        CopierSrcAcc(Ctx.Buf1, CGH);
+    S::accessor<int, 1, S::access::mode::write,
+                S::access::target::host_buffer>
+        CopierDstAcc(Ctx.Buf2, CGH);
+
+    auto CopierHostTask = [CopierSrcAcc, CopierDstAcc, &Ctx]() {
+      for (size_t Idx = 0; Idx < CopierDstAcc.get_count(); ++Idx)
+        CopierDstAcc[Idx] = CopierSrcAcc[Idx];
+
+      bool Expected = false;
+      bool Desired = true;
+      assert(Ctx.Flag.compare_exchange_strong(Expected, Desired));
+
+      // let's employ some locking here
+      {
+        std::lock_guard<std::mutex> Lock(Ctx.Mutex);
+        Ctx.CV.notify_all();
+      }
+    };
+
+    CGH.codeplay_host_task(CopierHostTask);
+  });
+}
+
+void Thread2Fn(Context &Ctx) {
+  std::unique_lock<std::mutex> Lock(Ctx.Mutex);
+
+  // T2.1. Wait until flag F is set eq true.
+  Ctx.CV.wait(Lock, [&Ctx] { return Ctx.Flag.load(); });
+
+  assert(Ctx.Flag.load());
+
+  // T2.2. print some "hello, world" message
+  Ctx.Message = "Hello, world";
+}
+
+void test() {
+  auto EH = [](S::exception_list EL) {
+    for (const std::exception_ptr &E : EL) {
+      throw E;
+    }
+  };
+
+  S::queue Queue(EH);
+
+  Context Ctx{{false}, Queue, "", {10}, {10}};
+
+  // 0. setup: thread 1 T1: exec smth; thread 2 T2: waits; init flag F = false
+  std::thread Thread1(Thread1Fn, std::reference_wrapper<Context>(Ctx));
+  std::thread Thread2(Thread2Fn, std::reference_wrapper<Context>(Ctx));
+
+  Thread1.join();
+  Thread2.join();
+
+  assert(Ctx.Flag.load());
+  assert(Ctx.Message == "Hello, world");
+
+  // 3. check via host accessor that buf 2 contains valid data
+  {
+    S::accessor<int, 1, S::access::mode::read,
+                S::access::target::host_buffer>
+        ResultAcc(Ctx.Buf2);
+
+    for (size_t Idx = 0; Idx < ResultAcc.get_count(); ++Idx) {
+      assert(ResultAcc[Idx] == Idx && "Invalid data in result buffer");
+    }
+  }
+}
+
+int main() {
+  test();
+
+  return 0;
+}
```

Comment:
Are there any differences in the tests?

Reply:
Yes, there is a difference. host-task-dependency is the same as host-task but with another kernel added after the host-task.

Reply:
This test is removed now.